x-pack/plugins/observability_solution/inventory/server/routes/entities/get_entity_groups.ts

```diff
@@ -31,8 +31,14 @@ export async function getEntityGroupsBy({
   const limit = `LIMIT ${MAX_NUMBER_OF_ENTITIES}`;
   const query = [from, ...where, group, sort, limit].join(' | ');
 
-  return inventoryEsClient.esql<EntityGroup>('get_entities_groups', {
-    query,
-    filter: esQuery,
-  });
+  const { hits } = await inventoryEsClient.esql<EntityGroup, { transform: 'plain' }>(
```

Comment:
why is here `plain`?

Reply:
hmm yeah I'll change to unflatten, I can't tell from the type whether unflattening was necessary but it probably is.

Reply:
Actually, using `unflatten` here leads to surprises and incompatibility with the type: `{ [x:string]: string }` becomes `{ [x:string]: string | { [x:string]: string }}`, so a value like `my.type` would become `{ my: { type: string } }`. I don't think the code expects that. I'll leave `plain` here.